test/functional/pdo_sqlsrv/pdoStatement_bindParam_inout_emulate_prepare.phpt

```diff
@@ -1,30 +1,36 @@
 --TEST--
 Tests error returned when binding input/output parameter with emulate prepare
 --SKIPIF--
-<?php require('skipif.inc'); ?>
+<?php require_once('skipif_mid-refactor.inc'); ?>
 --FILE--
 <?php
-require_once("MsSetup.inc");
-$dsn = "sqlsrv:Server=$server ; Database = $databaseName";
+require_once("MsCommon_mid-refactor.inc");
 try {
-    $dbh = new PDO($dsn, $uid, $pwd);
-    $dbh->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+    $dbh = connect();
 
-    $dbh->query("IF OBJECT_ID('sp_ReverseString', 'P') IS NOT NULL DROP PROCEDURE sp_ReverseString");
-    $dbh->query("CREATE PROCEDURE sp_ReverseString @String as VARCHAR(2048) OUTPUT as SELECT @String = REVERSE(@String)");
-    $stmt = $dbh->prepare("EXEC sp_ReverseString ?", array(PDO::ATTR_EMULATE_PREPARES => true));
+    $procName = 'sp_ReverseString';
+    dropProc($dbh, $procName);
+    $dbh->query("CREATE PROCEDURE $procName @String as VARCHAR(2048) OUTPUT as SELECT @String = REVERSE(@String)");
+    $stmt = $dbh->prepare("EXEC $procName ?", array(PDO::ATTR_EMULATE_PREPARES => true));
     $string = "123456789";
     $stmt->bindParam(1, $string, PDO::PARAM_STR | PDO::PARAM_INPUT_OUTPUT, 2048);
     $stmt->execute();
-    print "Result: ".$string;
+    print "Result: $string";
 
     //free the statement and connection
-    $stmt = null;
-    $dbh = null;
-}
-catch(PDOException $e) {
-    print("Error: " . $e->getMessage() . "\n");
+    dropProc($dbh, $procName);
+    unset($stmt);
+    unset($dbh);
+} catch (PDOException $e) {
+    $error = $e->getMessage();
+    $pass = !isColEncrypted() && $error === "SQLSTATE[IMSSP]: Statement with emulate prepare on does not support output or input_output parameters.";
+    $pass |= isColEncrypted() && ($error === "SQLSTATE[IMSSP]: Connection with Column Encryption enabled do no support PDO::ATTR_EMULATE_PREPARES with binding parameters.");
```

Comment:
I didn't realize that the error message is not right... shouldn't it be "does not support"?

Reply:
Checking error messages calls `isColEncrypted()` twice. Is it necessary to do it this way? 
Should this be isAEConnect() instead?

Reply:
isAEConnected is very quick (just checking the $keystore isn't null), so not going to change the code to only calling isAEConnected once.

Reply:
so would you change to isAEConnected()?